### PR TITLE
Mix Scala 2 macros in MUnit Scala 3 module

### PR DESCRIPTION
**Pull request by olafurpg**
With this change, Scala 2.13 users will be able to depend on munit_3.0
artifacts instead of munit_2.13.

I wasn't able to manually test this change because MUnit doesn't
cross-build for 3.0.0-M1, which is the latest version supported by the
`-Ytasty-reader` flag in 2.13.4. Once 2.13.5 is out, it will be easier
ot make use of this feature.

### CODE REVIEW

File: website/blog/2021-01-05-macromix.md
Comment (olafurpg) [inline website/blog/2021-01-05-macromix.md:8]: cc/ @bishabosha @sjrd could you please take a look at this blog post? Please correct me if I got the details wrong

Comment (olafurpg) [reply website/blog/2021-01-05-macromix.md:8]: Live preview at https://olafurpg.github.io/munit/blog/2021/01/05/macromix.html so you can see the rendered SVG files.

**Review by gabro — COMMENTED** [review]
Really cool!

What's the longer plan though? Do we still publish MUnit also for 2.13? If not, I assume Scala 2 users will need to use the full artifact name in sbt to depend on the 3.0 artifact? Either way it may be good to add some notes about this in the user documentation, what do you think?

File: build.sbt
Comment (gabro) [inline build.sbt:188]: Does isDotty come from the sbt-dotty plug-in? Maybe we could use isScala3 for consistency with the rest of the build

File: website/blog/2021-01-05-macromix.md
Comment (bishabosha) [inline website/blog/2021-01-05-macromix.md:175]: I was confused to find that I could compile a macro bundle using Scala 3.0.0-M1 but the macro expansion failed from Scala 2.13.4, despite the macro bundle having the same bytecode when compiled by either 2.13.4 or 3.0.0-M1, and the tasty to call the macro was the same either way. Something to do with reflective invocation not having the required number of arguments.

It's probably worth opening an issue with the trouble you found, because I don't see at this moment why it should fail.

my example was the following:
```scala
class TpeTagImpl(val c: Context) {
  import c.universe._

  // factory for `final case class TpeTag[T](repr: String)`
  // invoked by `implicit def mkTpeTag[T]: TpeTag[T] = macro TpeTagImpl.mkTag[T]`
  def mkTag[T](implicit T: c.WeakTypeTag[T]): Tree = {
    val tpeString = Literal(Constant(T.tpe.toString))
    // new testframework.TpeTag[T](T.tpe.toString)
    New(appliedType(c.mirror.staticClass(classOf[testframework.TpeTag[_]].getName()).toType, T.tpe), tpeString)
  }

}
```

File: build.sbt
Comment (olafurpg) [reply build.sbt]: Good point. Fixed.

**Comment by olafurpg** [comment]
@gabro thank you for the review!

> What's the longer plan though? Do we still publish MUnit also for 2.13? If not, I assume Scala 2 users will need to use the full artifact name in sbt to depend on the 3.0 artifact? Either way it may be good to add some notes about this in the user documentation, what do you think?

I was thinking of keeping things unchanged. I updated the post to explain that MUnit 2.13 will continue to be published.

The primary purpose of this post is to advocate for this route since I don't everybody are fully aware of the risks of cross-building 2.13 and 3.

**Review by bishabosha — APPROVED** [review]

File: website/blog/2021-01-05-macromix.md
Comment (olafurpg) [reply website/blog/2021-01-05-macromix.md]: I removed the macro bundle mention. I didn't try using bundles, I just assumed they wouldn't work because I thought they were implemented with macroparadise.

Comment (olafurpg) [reply website/blog/2021-01-05-macromix.md]: I confused bundles with https://github.com/milessabin/macro-compat, which is implemented as a macro annotation to add bundle support in Scala 2.11

**Review by sjrd — APPROVED** [review]
A bit late, but FWIW the blog post LGTM.

**Comment by olafurpg** [comment]
@sjrd whoops 😅 I should have waited for your LGTM before merging.

File: website/blog/2021-01-05-macromix.md
Comment (bishabosha) [reply website/blog/2021-01-05-macromix.md]: This is very nice work, I will open the issue then for my example.

**Comment by sjrd** [comment]
No worries. I could have looked at this earlier this morning. ;)

**Pull request merged**
